apps/survey/src/components/handlers/standard/SameAsBeforePromptHandler.vue

```diff
@@ -27,15 +28,59 @@ const {
 const survey = useSurvey();
 
 const sabFood = useSameAsBefore().getItem(survey.localeId, code);
+const updatedSabFood = {
+  ...(sabFood ?? {}),
+  food: {
+    ...(sabFood?.food ?? {}),
+    portionSize: sabFood?.food?.portionSize ? { ...sabFood.food.portionSize } : undefined,
+    portionSizeMethodIndex: sabFood?.food?.portionSizeMethodIndex ?? undefined,
+    linkedFoods: sabFood?.food?.linkedFoods ? [...sabFood.food.linkedFoods] : [],
+    customPromptAnswers: sabFood?.food?.customPromptAnswers ? { ...sabFood.food.customPromptAnswers } : {},
+    flags: sabFood?.food?.flags ? [...sabFood.food.flags] : [],
+  },
+};
+function onSabOptionsUpdate(sabOptions: Record<string, boolean>): void {
+  console.debug('Received sabOptions from child:', sabOptions);
+  if (!sabOptions.portionSize && updatedSabFood?.food?.portionSize) {
+    updatedSabFood.food.portionSizeMethodIndex = undefined;
+    updatedSabFood.food.portionSize = undefined;
+    console.debug('PortionSize and Portion method index is removed as SAB prompt checkbox is explicitly set to false');
+    if (Array.isArray(updatedSabFood.food.flags)) {
+      updatedSabFood.food.flags = updatedSabFood.food.flags.filter(
+        (flag: string) =>
+          flag !== 'portion-size-option-complete'
```

Comment:
simplify to `[...your flaglist].includes(flag)` to avoid multiple lengthy conditions